MapboxNavigation/RouteMapViewController.swift

```diff
@@ -134,73 +152,110 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
             break
         }
     }
-    
+
+    func updateVisibleBounds(coordinates: [CLLocationCoordinate2D]) {
+        let camera = mapView.camera
+        camera.pitch = 0
+        camera.heading = 0
+        mapView.camera = camera
+
+        let polyline = MGLPolyline(coordinates: coordinates, count: UInt(coordinates.count))
+        // Don't keep zooming in
+        guard polyline.overlayBounds.ne - polyline.overlayBounds.sw > 200 else {
+            return
+        }
+        mapView.setVisibleCoordinateBounds(polyline.overlayBounds, edgePadding: overviewContentInset, animated: true)
```

Comment:
FYI, the map SDK has a "targeted course tracking" feature that accomplishes a similar effect. You keep the userTrackingMode as followWithCourse but also set targetCoordinate. The map automatically adjusts the visible coordinate bounds to keep both the current location and target coordinate in view at the same time. What's nice is that the user puck continues to show (and point in the right direction).

Nevertheless, I think this code is fine for now, because targeted course tracking mode doesn't reset to north (which is arguably better) and may not show the entire route line (if part of the route bows way off to the side or behind the user.